app/models/skin.rb

```diff
@@ -276,12 +279,14 @@ def get_all_parents
   
   # This is the main function that actually returns code to be embedded in a page
   def get_style(roles_to_include = DEFAULT_ROLES_TO_INCLUDE)
-    style = ""
-    if self.get_role != "override" && self.get_role != "site"
-      style += AdminSetting.default_skin != Skin.default ? AdminSetting.default_skin.get_style(roles_to_include) : (Skin.get_current_site_skin ? Skin.get_current_site_skin.get_style(roles_to_include) : '')
+    Rails.cache.fetch('Skins_html/v1/'+self.get_role+'/'+self.id.to_s+'/'+AdminSetting.default_skin.updated_at.to_s+'/'+Skin.default.updated_at.to_s+'/'+Skin.get_current_version+'/'+(Rails.cache.fetch('skins_generation') || '0')) do 
```

Comment:
It took me a minute to figure out what `Skins_html` was, since skins themselves are CSS. Could we change the name to something a bit clearer (and a bit more lowercase), e.g. style_tags or skins_style_tags? (Assuming that's what's being cached here -- the `<style type="css/text"></style>` bit?)